### PR TITLE
update golangci-lint to 1.46.2

### PR DESCRIPTION
**Pull request by systay**
The title is not lying

### CODE REVIEW

**Comment by systay** [comment]
> LGTM! Let's change the version in `.golangci.yml` too

ah... it was already not in sync with the other two settings, that's why I missed it. I fix

File: go/tools/release-notes/release_notes.go [hunk 1]
```diff
@@ -495,7 +495,7 @@ func main() {
 	flag.Parse()
 
 	// The -version flag must be of a valid format.
-	rx := regexp.MustCompile("v([0-9]+)\\.([0-9]+)\\.([0-9]+)")
+	rx := regexp.MustCompile(`v(\d+)\.(\d+)\.(\d+)`)
```
Comment (GuptaManan100) [inline go/tools/release-notes/release_notes.go:498]: This looks like it is changing the regex? I see we have removed `\` characters. The `\d` change if fine. But we are removing `\` before `.` as well. Maybe I am wrong.

File: go/vt/grpcoptionaltls/server_test.go [hunk 2]
```diff
@@ -110,7 +112,7 @@ func TestOptionalTLS(t *testing.T) {
 
 	t.Run("Plain2TLS", func(t *testing.T) {
 		for i := 0; i < 5; i++ {
-			testFunc(t, grpc.WithInsecure())
+			testFunc(t, grpc.WithTransportCredentials(insecure.NewCredentials()))
```
[the remark this answers is not in the document]
Comment (systay) [reply go/vt/grpcoptionaltls/server_test.go:115]: Neither do I.

```
go/vt/grpcclient/client.go:131:10: SA1019: grpc.WithInsecure is deprecated: use WithTransportCredentials and insecure.NewCredentials() instead. Will be supported throughout 1.x. (staticcheck)
```

File: go/tools/release-notes/release_notes.go (hunk 1 above)
Comment (systay) [reply go/tools/release-notes/release_notes.go:498]: Two different changes in one go. One was `[0-9]` -> `\d`, which you correctly identified.

The other one was:

```
"v([0-9]+)\\.([0-9]+)\\.([0-9]+)" -> `v([0-9]+)\.([0-9]+)\.([0-9]+)`
```

Makes sense?

File: go/vt/grpcoptionaltls/server_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/grpcoptionaltls/server_test.go:115]: Hmmm, okay 👍

File: go/tools/release-notes/release_notes.go (hunk 1 above)
Comment (GuptaManan100) [reply go/tools/release-notes/release_notes.go:498]: oh okay, I missed the `"` to ` change

**Review by GuptaManan100 — APPROVED** [review]

File: go/test/endtoend/vtorc/utils/utils.go [hunk 3]
```diff
@@ -877,7 +877,7 @@ func AddSemiSyncKeyspace(t *testing.T, clusterInfo *VtOrcClusterInfo) {
 	}
 
 	vtctldClientProcess := cluster.VtctldClientProcessInstance("localhost", clusterInfo.ClusterInstance.VtctldProcess.GrpcPort, clusterInfo.ClusterInstance.TmpDirectory)
-	out, err := vtctldClientProcess.ExecuteCommandWithOutput("SetKeyspaceDurabilityPolicy", keyspaceSemiSyncName, fmt.Sprintf("--durability-policy=semi_sync"))
```
Comment (ajm188) [inline go/test/endtoend/vtorc/utils/utils.go:880]: lol

**Review by ajm188 — APPROVED** [review]

File: go/test/endtoend/vtorc/utils/utils.go (hunk 3 above)
Comment (systay) [reply go/test/endtoend/vtorc/utils/utils.go:880]: it made sense at the time!

**Pull request merged**